x-pack/libbeat/management/managerV2.go

```diff
@@ -590,7 +590,23 @@ func (cm *BeatV2Manager) reload(units map[unitKey]*client.Unit) {
 
 	// reload the output configuration
 	if err := cm.reloadOutput(outputUnit); err != nil {
-		errs = append(errs, err)
+		// Output creation failed, there is no point in going any further
+		// because there is no output read the events.
+		//
+		// Trying to start inputs will eventually lead them to deadlock
+		// waiting for the output. Log input will deadlock when starting,
+		// effectively blocking this manager.
+		err = fmt.Errorf("could not start output: %w", err)
+		outputUnit.UpdateState(client.UnitStateFailed, err.Error(), nil)
+		cm.status = lbmanagement.Failed
+		cm.message = err.Error()
+
+		// If there are any other errors, set the status accordingly.
+		if len(errs) > 0 {
+			errs = append(errs, err)
```

Comment:
This looks like we only append this error to the slice if the slice already has entries.  What happens if this is the first error, is there a reason not to append it?  It looks like original implementation did.

Reply:
Anyways I added an extra comment trying to make it more clear.